---
title: "Seurat Cell Categorization"
author: "Reha Mathur"
output: html_notebook
---
## importing packages 
```{r}
library("devtools")
install_github("jw156605/SLICER")
library(Seurat)
library(readr)
library(tidyverse)
library(HGC) 
library(latticeExtra)
library(lattice) 
library(RColorBrewer)
library(clusterProfiler)
library(org.Mm.eg.db)
library(AnnotationDbi)
```
## reading in normalized data again
```{r}
exp = read_csv("normalized_data.csv")
exp = discard(exp,~all(is.na(.) |.  == ""))
exp = distinct(exp, `...1`, .keep_all = TRUE)
# filtering out the reprogramming transcription factors 
exp = filter(exp, !`...1`%in% c("Mef2c", "Gata4", "Tbx5"))
exp = column_to_rownames(exp, var="...1")

```
## creating a seurat object with all of the cells 

```{r}
seurat.obj = CreateSeuratObject(counts = exp)
```
```{r}
# using seurat to noramlize, scale and dimensionally reduce the data 
seurat.obj = NormalizeData(seurat.obj)
seurat.obj = FindVariableFeatures(seurat.obj, selection.method =  "vst", nfeatures = 10000)
all.genes = rownames(seurat.obj)
seurat.obj = ScaleData(seurat.obj, features = all.genes)
seurat.obj = RunPCA(seurat.obj, features =  VariableFeatures(seurat.obj))

```
```{r}
# elbow plot to see the correct number of dimensions
ElbowPlot(seurat.obj)
```
## Seurat clustering the entire graph to see what clusters are identified 
```{r}
seurat.obj = FindNeighbors(seurat.obj, dims = 1:15)
```

```{r}
seurat.obj = FindClusters(seurat.obj, resolution = 1.75)
```
## importing the results from the paper and our hierarchical clustering results for further comparison
```{r}
identified_cells = read.csv("paper_cell_states.csv", row.names = 1, header=TRUE)
```
```{r}
identified_cells = identified_cells[match(rownames(seurat.obj@meta.data), rownames(identified_cells)),]
seurat.obj@meta.data$cc_paper = identified_cells$CC
seurat.obj@meta.data$cat_paper = identified_cells$stage
seurat.obj@meta.data$cc_hclust = "CCI"
seurat.obj@meta.data[active_cells,]$cc_hclust = "CCA"

```


```{r}
Idents(seurat.obj)  = "RNA_snn_res.1.75"
```
## running UMAP dimensional reduction on the data to plot the data 
```{r}
seurat.obj = RunUMAP(seurat.obj, dims = 1:15)
```
# seeing what tsne reduction makes the data do 
```{r}
seurat.obj = RunTSNE(seurat.obj, dims = 1:15)
DimPlot(seurat.obj, reduction = "tsne", group.by = 'RNA_snn_res.1.75')
``` 
# finding differentially expressed markers in these clusters
```{r}
markers = FindAllMarkers(seurat.obj, logfc.threshold =  0.25, min.pct = 0.1, only.pos = FALSE, test.use = 'MAST')
```
# top markers for the clusters of all of the cells and violin plots of their expression 
```{r}
top_markers = markers %>%
    group_by(cluster) %>%
    slice_max(n = 2, order_by = avg_log2FC)
```

```{r}
VlnPlot(seurat.obj, features = c("Rbp1", "Dhrs3"))
VlnPlot(seurat.obj, features = c("Fstl1", "Bgn"))
VlnPlot(seurat.obj, features = c("Serpinh1", "Dsp"))
VlnPlot(seurat.obj, features = c("Prc1", "Birc5"))
```
## categorizing the cells based off their cell cycle analysis 
```{r}
## cell cycle 
s.genes <- str_to_title(cc.genes$s.genes)
g2m.genes <- str_to_title(cc.genes$g2m.genes)

seurat.obj = CellCycleScoring(seurat.obj, s.features = s.genes, g2m.features = g2m.genes)
seurat.obj@meta.data$cat_cluster = cell_categorized$stage

```

## plotting the comparison of the cell cycle results
```{r}
clusters = DimPlot(seurat.obj, group.by = "RNA_snn_res.1.75", label = TRUE, reduction = "umap")
real_condition = DimPlot(seurat.obj, group.by = "cc_paper", label = TRUE, reduction = "umap") 
phase = DimPlot(seurat.obj, group.by = "Phase", label = TRUE, reduction = "umap") 
cat = DimPlot(seurat.obj, group.by = "cc_hclust", label = TRUE, reduction = "umap")

clusters|real_condition|cat|phase


```



## subsetting the cells that fit with hclust and performing clustering on each group 

```{r}
#cca_seurat = subset(x = seurat.obj, subset = cc_hclust == "CCA")
#cci_seurat = subset(x = seurat.obj, subset = cc_hclust == "CCI")
cca_seurat = subset(x = seurat.obj, subset = Phase == "S" | Phase == "G2M")
cci_seurat = subset(x = seurat.obj, subset = Phase == "G1")
```

## clustering CCI cels 

```{r}
cci_seurat = FindNeighbors(cci_seurat, dims = 1:15)
cci_seurat = FindClusters(cci_seurat, resolution = 1)
```

```{r}
cl= DimPlot(cci_seurat, group.by = "RNA_snn_res.1",label = TRUE)
pap = DimPlot(cci_seurat, group.by = "cat_paper", label = TRUE)
hgc = DimPlot(cci_seurat, group.by = "cat_cluster", label = TRUE)

cl|pap|hgc
```
## identifying CCI cells 

```{r}
cci_markers = FindAllMarkers(cci_seurat, logfc.threshold =  0.25, min.pct = 0.1, only.pos = FALSE, test.use = 'MAST')
```
```{r}
cci_top_markers = cci_markers %>%
    group_by(cluster) %>%
    slice_min(n = 5, order_by = p_val_adj)
```

```{r}
VlnPlot(cci_seurat, features = c("Fstl1", "Bgn"), cols = c("purple", "light green", "red", "blue"))
VlnPlot(cci_seurat, features = c("Tnni3", "Smpx"), cols = c("purple", "light green", "red", "blue"))
VlnPlot(cci_seurat, features = c("Vcam1", "Tm4sf1"), cols = c("purple", "light green", "red", "blue"))
VlnPlot(cci_seurat, features = c("Clca2", "Cd200"), cols = c("purple", "light green", "red", "blue"))
VlnPlot(cci_seurat, features = c("Lox", "Tm4sf1"), cols = c("purple", "light green", "red", "blue"))

```

Fstl1 involved in endothelial cell differentiation and endothelial cell migration and Bgn is a gene that is used for connective tissue metabolism -> lower expression -> cluster 1 = iCM, cluster 2 likely pICM

Tnni3, smpx, dsp used for cardiomycoytes -> cluster 2 = pre-iCM, cluster 1 and 3 are still fibroblasts 

```{r}
Idents(cci_seurat)  = "RNA_snn_res.1"
cci_seurat = RenameIdents(cci_seurat, `0` = "0")
cci_seurat = RenameIdents(cci_seurat, `1` = "iCM")
cci_seurat = RenameIdents(cci_seurat, `2` = "piCM")
cci_seurat = RenameIdents(cci_seurat, `3` = "3")

```

```{r}
cl= DimPlot(cci_seurat, label = TRUE, cols = c("purple", "light green", "red", "blue"))
pap = DimPlot(cci_seurat, group.by = "cat_paper", label = TRUE, cols = c("blue", "red", "purple", "light green"))
hgc = DimPlot(cci_seurat, group.by = "cat_cluster", label = TRUE, cols = c("blue", "red", "purple", "light green"))

cl|pap|hgc
```
## clustering CCA cells

```{r}
cca_seurat = FindNeighbors(cca_seurat, dims = 1:15)
cca_seurat = FindClusters(cca_seurat, resolution = 1)
```

```{r}
cl= DimPlot(cca_seurat, group.by = "RNA_snn_res.1",label = TRUE)
pap = DimPlot(cca_seurat, group.by = "cat_paper", label = TRUE)
cl|pap
```
## identifying CCA cells 

```{r}
cca_markers = FindAllMarkers(cca_seurat, logfc.threshold =  0.25, min.pct = 0.1, only.pos = FALSE, test.use = 'MAST')
```
```{r}
cca_top_markers = cca_markers %>%
    group_by(cluster) %>%
    slice_min(n = 5, order_by = p_val_adj)
```

```{r}
VlnPlot(cca_seurat, features = c("Fstl1", "Bgn"), cols = c("blue", "purple", "light green", "red"))
VlnPlot(cca_seurat, features = c("Tnni3", "Smpx"), cols = c("blue", "purple", "light green", "red"))
VlnPlot(cca_seurat, features = c("Vcam1", "Tm4sf1"), cols = c("blue", "purple", "light green", "red"))
VlnPlot(cca_seurat, features = c("Clca2", "Cd200"), cols = c("blue", "purple", "light green", "red"))
```
#### 

Tnni3 related to heart fucntion -> iCM= 2 

```{r}
Idents(cca_seurat)  = "RNA_snn_res.1"
cca_seurat = RenameIdents(cca_seurat, `2` = "iCM")
cca_seurat = RenameIdents(cca_seurat, `1` = "1")
cca_seurat = RenameIdents(cca_seurat, `0` = "0")
cca_seurat = RenameIdents(cca_seurat, `3` = "3")
```

```{r}
cl= DimPlot(cca_seurat, label = TRUE,cols = c("blue", "purple", "light green", "red"))
pap = DimPlot(cca_seurat, group.by = "cat_paper", label = TRUE, cols = c("blue", "red", "purple", "light green"))
hgc = DimPlot(cca_seurat, group.by = "cat_cluster", label = TRUE, cols = c("blue", "red", "purple", "light green"))
cl|pap|hgc

```

## Trying out SCTransform which is supposedly better for single cell data
```{r}
seurat.sc = CreateSeuratObject(counts = exp)
seurat.sc = SCTransform(seurat.sc)
```
```{r}
seurat.sc = RunPCA(seurat.sc, features =  VariableFeatures(seurat.sc))
```

```{r}
identified_cells = identified_cells[match(rownames(seurat.sc@meta.data), rownames(identified_cells)),]
seurat.sc@meta.data$cc_paper = identified_cells$CC
seurat.sc@meta.data$cat_paper = identified_cells$stage

seurat.sc = RunUMAP(seurat.sc, dims = 1:15)
seurat.sc = FindNeighbors(seurat.sc, dims = 1:15)
seurat.sc = FindClusters(seurat.sc, resolution = 1.75)
```

```{r}
seurat.sc = CellCycleScoring(seurat.sc, s.features = s.genes, g2m.features = g2m.genes)
```

```{r}
clusters = DimPlot(seurat.sc, group.by = "SCT_snn_res.1.75", label = TRUE, reduction = "umap")
real_condition = DimPlot(seurat.sc, group.by = "cc_paper", label = TRUE, reduction = "umap") 
phase = DimPlot(seurat.sc, group.by = "Phase", label = TRUE, reduction = "umap") 
cat = DimPlot(seurat.sc, group.by = "cat_paper", label = TRUE, reduction = "umap") 

clusters|real_condition|phase|cat

```

The results are largely the same so just going to continue on with the original clsutering. 